lms/djangoapps/courseware/views.py

```diff
@@ -658,6 +674,10 @@ def course_info(request, course_id):
     with modulestore().bulk_operations(course_key):
         course = get_course_with_access(request.user, 'load', course_key)
 
+        # check to see if there is entrance exam not passed then it redirect to courseware
+        if not has_access(request.user, 'view_courseware_with_entrance_exam', course):
```

Comment:
This access check could become a part of the proposed `lms\djangoapps\courseware\entrance_exams.py` module